src/v2/components/ConnectionSelectionList/components/SearchedChannels/index.tsx

```diff
@@ -1,33 +1,61 @@
-import React from 'react'
-import { useQuery } from '@apollo/client'
+import React, { useCallback, useEffect, useState } from 'react'
+import { useMutation, useQuery } from '@apollo/client'
+import { useKeyboardListNavigation } from 'use-keyboard-list-navigation'
 
 import Indicator from 'v2/components/ConnectionSelectionList/components/Indicator'
 import { ChannelsList } from 'v2/components/ConnectionSelectionList/components/ChannelsList'
-import { CreatePrivateChannelButton } from 'v2/components/ConnectionSelectionList/components/CreatePrivateChannelButton'
+import {
+  CreatePrivateChannelButton,
+  Mode,
+} from 'v2/components/ConnectionSelectionList/components/CreatePrivateChannelButton'
 import { OnConnectionSelectionType } from 'v2/components/ConnectionSelectionList'
+import { SelectableChannel as Channel } from '__generated__/SelectableChannel'
 
 import SEARCHED_CHANNELS_QUERY from 'v2/components/ConnectionSelectionList/components/SearchedChannels/queries/searchedChannels'
 import {
   SearchedChannelsQuery,
   SearchedChannelsQueryVariables,
 } from '__generated__/SearchedChannelsQuery'
 
+import {
+  createPrivateChannelMutation as CreatePrivateChannelMutation,
+  createPrivateChannelMutationVariables as CreatePrivateChannelMutationVariables,
+} from '__generated__/createPrivateChannelMutation'
+import createPrivateChannelMutation from 'v2/components/ConnectionSelectionList/components/CreatePrivateChannelButton/mutations/createPrivateChannel'
+
 interface SearchedChannelsProps {
   query: string
   onConnectionSelection: OnConnectionSelectionType
   includeOpenChannels: boolean
+  selectedChannels: Channel[]
+  searchRef: React.MutableRefObject<any>
 }
 
-export const SearchedChannelsResults: React.FC<SearchedChannelsProps> = ({
+interface SearchedChannelsResultsProps {
+  setChannels: React.Dispatch<React.SetStateAction<Channel[]>>
+  highlightedIndex: number
+}
+
+export const SearchedChannelsResults: React.FC<SearchedChannelsProps &
+  SearchedChannelsResultsProps> = ({
   query,
   onConnectionSelection,
   includeOpenChannels,
+  setChannels,
+  selectedChannels,
+  highlightedIndex,
 }) => {
   const { data, error, loading } = useQuery<
     SearchedChannelsQuery,
     SearchedChannelsQueryVariables
   >(SEARCHED_CHANNELS_QUERY, { variables: { query, includeOpenChannels } })
 
+  useEffect(() => {
+    if (data) {
+      setChannels([null, ...data.me.searched_channels])
```

Comment:
Why is there a null first item?

Reply:
Stands in for the create private channel button

Reply:
Would be worth adding a comment I reckon.